gossip/peer.go

```diff
@@ -85,7 +92,29 @@ func (p *peer) SetProgress(x PeerProgress) {
 	p.Lock()
 	defer p.Unlock()
 
+	// Check if the peer is progressing
+	if x.More(p.progress) {
+		p.setPeerAsProgressing(x)
+	}
+}
+
+func (p *peer) setPeerAsProgressing(x PeerProgress) {
 	p.progress = x
+	p.progressTime = time.Now()
```

Comment:
Any specific reason why `p.appMessageTime` is locked, but `p.progressTime` isn't?

Reply:
It's locked in `SetProgress()` where `setPeerAsProgressing()` is called.